ansible/roles/test/files/tools/loganalyzer/loganalyzer.py

```diff
@@ -219,28 +220,30 @@ def wait_for_marker(self, marker, timeout=120, polling_interval=10):
         last_check_pos = 0
         syslog_file = "/var/log/syslog"
         prev_syslog_file = "/var/log/syslog.1"
+        last_dt = prev_syslog_file
         while wait_time <= timeout:
-            # look for marker in syslog file
-            if os.path.exists(syslog_file):
-                with open(syslog_file, 'r') as fp:
-                    # resume from last search position
-                    if last_check_pos:
-                        fp.seek(last_check_pos)
-                    # check if marker in the file
-                    for logs in fp:
-                        if marker in logs:
-                            return True
-                    # record last search position
-                    last_check_pos = fp.tell()
-
-            # logs might get rotated while waiting for marker
-            # look for marker in syslog.1 file
-            if os.path.exists(prev_syslog_file):
-                with open(prev_syslog_file, 'r') as pfp:
-                    # check if marker in the file
-                    for logs in pfp:
-                        if marker in logs:
-                            return True
+            with open(syslog_file, 'r') as fp:
+                dt = os.path.getctime(syslog_file)
+                if last_dt != dt:
+                    try:
+                        with open(prev_syslog_file, 'r') as pfp:
+                            pfp.seek(last_check_pos)
+                            for log in fp:
+                                if marker in log:
+                                    return True
+                    except OSError:
```

Comment:
@mannytaheri can you share why these changes are needed here in loganalyzer ? This is common code which gets used for non vsonic tests also.